src/Spectre.libWavelet.Tests/WaveletDecomposerRefTest.cpp

```diff
@@ -0,0 +1,171 @@
+                0.0633207123092398,
+                -0.0651742255517986,
+                -0.0212504983010336,
+                -1.63488447864066e-06
+            };
+            firstHighFrequencyResult = {
+                0,
+                -0.230377813308897,
+                0.254090943935123,
+                0.107678933249283,
+                -0.0667168468534168,
+                -0.0540778152370236,
+                -0.0105974017850692,
+                -2.22044604925031e-16,
+                -2.22044604925031e-16,
+                -8.88178419700125e-16,
+                2.30377813308897,
+                -4.61430975913130,
+                1.21002916292327,
+                1.63627886777771,
+                -0.0596734693105170,
+                -0.380726319282519,
+                -0.0953766160656215
+            };
+            firstLowFrequencyResult = {
+                0.0,
+                -4.06844404044861e-07,
+                2.69624376778159e-05,
+                0.000170805548639498,
+                -0.00170291259939498,
+                0.00309960587361391,
+                -0.0165956761940691,
+                0.0479600502210081,
+                0.0561611188212727,
+                -0.339939690673109,
+                -0.0487037936781232,
+                1.23299325679310,
+                1.41702876032074,
+                0.461966379081157,
+                3.55408918941385e-05,
+                0.0
+            };
+            lastLowFrequencyResult = {
+                -4.64283404088808e-07,
+                2.90564870809657e-05,
+                0.000170041467736160,
+                -0.00176998109340235,
+                0.00336519663747599,
+                -0.0167679988715969,
+                0.0469668204068491,
+                0.0577080776967957,
+                -0.338551325051386,
+                -0.0523643871799257,
+                1.23076472595957,
+                1.41934784856588,
+                0.463535004863161,
+                6.73843951653069e-05,
+                0.0,
+                0.0
+            };
+        }
+
+        void InitializeSignal(Signal& signal)
+        {
+            for (unsigned i = 0; i < signal.size(); i++)
+            {
+                signal[i] = (DataType)i;
+            }
+        }
+
+    protected:
+        WaveletDecomposerRef decomposer;
+        std::vector<DataType> lastRowLastHighFrequencyResult;
+        std::vector<DataType> firstRowLastHighFrequencyResult;
+        std::vector<DataType> firstHighFrequencyResult;
+        std::vector<DataType> firstLowFrequencyResult;
+        std::vector<DataType> lastLowFrequencyResult;
+    };
+
+    TEST_F(WaveletDecomposerRefTest, decomposes_the_signal)
+    {
+        Signal signal(10);
+        InitializeSignal(signal);
+        WaveletCoefficients coefficients = decomposer.Decompose(std::move(signal));
+
+        // Due to high amount of coefficients generated, only a few sets will be
```

Comment:
Again unnecessary split when code lines are wider than unsplit comment would be, makes it less transparent